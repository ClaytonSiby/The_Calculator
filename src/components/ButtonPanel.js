import PropTypes from 'prop-types';
import Button from './Button';

const ButtonPanel = ({ clickHandler }) => {
  const handleClick = btnName => clickHandler(btnName);
  return (
    <div>
      <div className="group-one">
        <Button buttonName="AC" clickHandler={handleClick} />
        <Button buttonName="+/-" clickHandler={handleClick} />
        <Button buttonName="%" clickHandler={handleClick} />
        <Button buttonName="รท" clickHandler={handleClick} />
      </div>

      <div className="group-two">
        <Button buttonName="7" clickHandler={handleClick} />
        <Button buttonName="8" clickHandler={handleClick} />
        <Button buttonName="9" clickHandler={handleClick} />
        <Button buttonName="X" clickHandler={handleClick} />
      </div>

      <div className="group-three">
        <Button buttonName="4" clickHandler={handleClick} />
        <Button buttonName="5" clickHandler={handleClick} />
        <Button buttonName="6" clickHandler={handleClick} />
        <Button buttonName="-" clickHandler={handleClick} />
      </div>

      <div className="group-4">
        <Button buttonName="1" clickHandler={handleClick} />
        <Button buttonName="2" clickHandler={handleClick} />
        <Button buttonName="3" clickHandler={handleClick} />
        <Button buttonName="+" clickHandler={handleClick} />
      </div>

      <div className="group-5">
        <Button buttonName="O" clickHandler={handleClick} />
        <Button buttonName="." clickHandler={handleClick} />
        <Button buttonName="=" clickHandler={handleClick} />
      </div>
    </div>
  );
};

ButtonPanel.propTypes = {
  clickHandler: PropTypes.func.isRequired,
};

export default ButtonPanel;
