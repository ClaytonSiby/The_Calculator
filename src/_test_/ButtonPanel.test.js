import {render} from '@testing-library/react';
import ButtonPanel from '../components/ButtonPanel';
import renderer from 'react-test-renderer';

describe('ButtonPanel Component', () => {
  test('renders successfully without raising errors', () => {
    render(<ButtonPanel />);
  });

  test('output renders consistently', () => {
    const test = renderer.create(<ButtonPanel />).toJSON();
    expect(test).toMatchSnapshot();
  })
});
