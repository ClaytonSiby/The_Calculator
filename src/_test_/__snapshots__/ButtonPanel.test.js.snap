// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonPanel Component output renders consistently 1`] = `
<div
  className="flexContainer"
>
  <div
    className="fixedHeight"
  >
    <button
      className="maxWidth"
      onClick={[Function]}
      style={
        Object {
          "background": false,
          "width": "25%",
        }
      }
      type="button"
    >
      AC
    </button>
    <button
      className="maxWidth"
      onClick={[Function]}
      style={
        Object {
          "background": false,
          "width": "25%",
        }
      }
      type="button"
    >
      +/-
    </button>
    <button
      className="maxWidth"
      onClick={[Function]}
      style={
        Object {
          "background": false,
          "width": "25%",
        }
      }
      type="button"
    >
      %
    </button>
    <button
      className="maxWidth"
      onClick={[Function]}
      style={
        Object {
          "background": "orange",
          "width": "25%",
        }
      }
      type="button"
    >
      รท
    </button>
  </div>
  <div
    className="fixedHeight"
  >
    <button
      className="maxWidth"
      onClick={[Function]}
      style={
        Object {
          "background": false,
          "width": "25%",
        }
      }
      type="button"
    >
      7
    </button>
    <button
      className="maxWidth"
      onClick={[Function]}
      style={
        Object {
          "background": false,
          "width": "25%",
        }
      }
      type="button"
    >
      8
    </button>
    <button
      className="maxWidth"
      onClick={[Function]}
      style={
        Object {
          "background": false,
          "width": "25%",
        }
      }
      type="button"
    >
      9
    </button>
    <button
      className="maxWidth"
      onClick={[Function]}
      style={
        Object {
          "background": "orange",
          "width": "25%",
        }
      }
      type="button"
    >
      x
    </button>
  </div>
  <div
    className="fixedHeight"
  >
    <button
      className="maxWidth"
      onClick={[Function]}
      style={
        Object {
          "background": false,
          "width": "25%",
        }
      }
      type="button"
    >
      4
    </button>
    <button
      className="maxWidth"
      onClick={[Function]}
      style={
        Object {
          "background": false,
          "width": "25%",
        }
      }
      type="button"
    >
      5
    </button>
    <button
      className="maxWidth"
      onClick={[Function]}
      style={
        Object {
          "background": false,
          "width": "25%",
        }
      }
      type="button"
    >
      6
    </button>
    <button
      className="maxWidth"
      onClick={[Function]}
      style={
        Object {
          "background": "orange",
          "width": "25%",
        }
      }
      type="button"
    >
      -
    </button>
  </div>
  <div
    className="fixedHeight"
  >
    <button
      className="maxWidth"
      onClick={[Function]}
      style={
        Object {
          "background": false,
          "width": "25%",
        }
      }
      type="button"
    >
      1
    </button>
    <button
      className="maxWidth"
      onClick={[Function]}
      style={
        Object {
          "background": false,
          "width": "25%",
        }
      }
      type="button"
    >
      2
    </button>
    <button
      className="maxWidth"
      onClick={[Function]}
      style={
        Object {
          "background": false,
          "width": "25%",
        }
      }
      type="button"
    >
      3
    </button>
    <button
      className="maxWidth"
      onClick={[Function]}
      style={
        Object {
          "background": "orange",
          "width": "25%",
        }
      }
      type="button"
    >
      +
    </button>
  </div>
  <div
    className="fixedHeight"
  >
    <button
      className="maxWidth"
      onClick={[Function]}
      style={
        Object {
          "background": false,
          "width": "50%",
        }
      }
      type="button"
    >
      0
    </button>
    <button
      className="maxWidth"
      onClick={[Function]}
      style={
        Object {
          "background": false,
          "width": "25%",
        }
      }
      type="button"
    >
      .
    </button>
    <button
      className="maxWidth"
      onClick={[Function]}
      style={
        Object {
          "background": "orange",
          "width": "25%",
        }
      }
      type="button"
    >
      =
    </button>
  </div>
</div>
`;
